### PR TITLE
added Google Drive support

### PR DESCRIPTION
**Pull request by iFun**

### CODE REVIEW

File: components/d2l-sequences-content-link-googledrive.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	static get properties() {
+		return {
+			href: {
+				type: String,
+				reflectToAttribute: true,
+				notify: true,
+				observer: '_scrollToTop'
+			},
+			previousHref: {
```
Comment (mpharoah-d2l) [inline components/d2l-sequences-content-link-googledrive.js]: Why is this here?

File: components/d2l-sequences-content-router.js [hunk 2]
```diff
@@ -95,8 +96,11 @@ function getLinkEntityType(linkActivity) {
 	const link = linkActivity.getLinkByRel('about');
 	const openInNewTab = linkActivity.hasClass('open-in-new-tab');
 
-	if (link && link.href.startsWith(window.location.protocol)) {
+	if (link && link.href.includes('docs.google')) {
```
Comment (mpharoah-d2l) [inline components/d2l-sequences-content-router.js]: This will miss some links formats and can have false positives. Here's a better check we use elsewhere:
```
const googleDriveRegex = /^https?:\/\/(?:docs|drive)\.google\.com(?:\/a\/[^\/]+)?\/(?!forms\/)[^\/]+\/d\/([^\/\?#]{2,})(?:$|[\/\?#])/;
const alternateLinkFormatRegex = /^https?:\/\/(?:docs|drive)\.google\.com\/open\?id=([^&#]+)(?:$|[&#])/;
const isGoogleDriveLink = googleDriveRegex.exec( url ) || alternateLinkFormatRegex.exec( url );
```

Comment (jwong-d2l) [reply components/d2l-sequences-content-router.js]: any chance we could put this regex in the HM endpoint and add a class instead that marks it as google docs?

Comment (mpharoah-d2l) [reply components/d2l-sequences-content-router.js]: Wish granted
https://git.dev.d2l/projects/CORE/repos/le/pull-requests/15002/overview

File: components/d2l-sequences-content-link-googledrive.js (hunk 1 above)
Comment (iFun) [reply components/d2l-sequences-content-link-googledrive.js]: I will remove it
